dasar/express-js/middleware.md

```diff
@@ -3,21 +3,19 @@
 ![middleware](middleware.png)
 
 ## 1. Permasalahan
+Pada Web Server yang sudah dibuat sebelumnya kita sudah dapat menginputkan index dan data yang akan dimanipulasi dari client.Namun server belum melakukan validasi apakah data yang dikirim client sudah lengkap atau belum.
 
-Biasanya, sebelum request dari client sampai ke tujuan. Perlu dilakukan sesuatu terhadap request tersebut. Misalnya autentikasi, sehingga hanya request yang sudah tervalidasi yang dapat dilanjutkan ke tujuan
+Kita dapat memberikan ( if ) pada setiap rute untuk melakukan validasi, namun hal ini akan membuat source code kita banyak, karena akan ada source code untuk melakukan validasi input pada setiap rute.
 
-## 2. Penjelasan Middleware
-
-Middleware merupakan fungsi yang dijalankan sebelum suatu request mencapai endpoint tujuan
+## 2. Penjelasan Materi Sebagai Solusi
```

Comment:
ini kata - kata di judulnya gak perlu dibuat sama kyk di issue nya sih, bisa langsung to the point aja bakal jelasin apa

```suggestion
## 2. Penjelasan Middleware
```